File: src/backend/php/app/Repositories/FileRepository.php
<?php

namespace App\Repositories;

use App\Models\File;
use App\Repositories\Contracts\FileRepositoryInterface;

/**
 * Eloquent-backed file repository.
 * Analogous to EF Core FileRepository : BaseRepository<File>.
 */
class FileRepository extends BaseRepository implements FileRepositoryInterface
{
    public function __construct(File $model)
    {
        parent::__construct($model);
    }
}

File: src/backend/python/api/repositories/file_repository.py
from sqlalchemy.orm import Session

from shared.models.file import File
from api.repositories.base_repository import BaseRepository, IRepository


class IFileRepository(IRepository[File]):
    """Contract for the file repository - mirrors an interface in C#."""


class FileRepository(BaseRepository[File], IFileRepository):
    """SQLAlchemy-backed file repository (analogous to EF Repository implementation)."""

    def __init__(self, db: Session) -> None:
        super().__init__(db, File)
